### ISSUE TITLE
How to use one set of optimiser parameters per hidden layer in a Chain?

### ISSUE THREAD

**Issue by rederekt** [FluxML/Flux.jl#715]
Right now, each Optimiser accepts a single Float64 for each hyperparameter. It would be useful to also accept an array of Float64 for a vector, where hidden layer `i` would use the hyperparameter at index `i`. This would be useful in the case of training additional hidden layers on top of a "backbone" network, where you'd want a lower learning rate for the pre-trained portion and higher learning rates for the new layers.

Is this capability already possible, is it in the works, or is it something that needs a PR?

**Comment by chengchingwen** [comment]
Since the Optimisers are mutable type on way to do this is to modify the learning rate and pass different Params each time. For example:

```julia
p1 = params(model1)
p2 = params(model2)
p = [ p1, p2 ]
lrate = [ 1e-5, 4e-4 ]

for (pi, lr) in zip(p, lrate)
  opt.eta = lr
  update!(opt, pi, grad)
end



```

**Comment by rederekt** [comment]
Gotcha, makes sense. Is there any reason why this functionality isn't a standard part of the library?

**Comment by rederekt** [comment]
I've also noticed [this line](https://github.com/FluxML/Flux.jl/blob/3a4c6274fadf1b94468da455a06f070b2cda6a64/src/optimise/train.jl#L64) in the documentation for `train!`:

"Multiple optimisers and callbacks can be passed to `opt` and `cb` as arrays."

That does not appear to be the case any more

**Comment by chengchingwen** [comment]
so let's say you are going to a chain of 10 dense layer, and every 2 layer would have a unique optimizer
```julia
model = Chain(Dense(...), ....,#the ten layer you want
              Dense(....))
ps = [params(model[(2i-1):(2i)]) for i = 1:5]
opts = [list of 5 different optimiser]
 #or can just be one and change learning rate each time
grad = gradient(()->loss, params(model))

for (opt, p) in zip(opts, ps)
    update!(opt, p, grad)
end
```

> I've also noticed this line in the documentation for train!:
 "Multiple optimisers and callbacks can be passed to opt and cb as arrays."

that part of code doesn't seems to exist anymore. It might be the old implementation,

**Comment by rederekt** [comment]
Ah, didn't realize the layers were indexable. My mistake, here's a more robust version of what I wanted  (one optimizer per layer), and can easily be changed to simply changing a parameter in an Optimizer:
```julia
m = Chain(Dense(2,4,tanh), Dense(4,1,tanh))
opts = [Descent(0.1), Descent(0.01)]

mult = [(params(l), opt) for (l, opt) in zip(m, opts)]

for (ps, opt) in mult
  Flux.train!(loss, ps, data, opt)
end
```
